src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -256,17 +264,26 @@ export const DiscoverGrid = ({
   /**
    * Pagination
    */
-  const [pagination, setPagination] = useState({ pageIndex: 0, pageSize: defaultPageSize });
+  const sampleRowsPerPage = useMemo(
```

Comment:
```suggestion
  const defaultRowsPerPage = useMemo(
```
Nit: I would consider 'default' or 'configured', etc. here instead of 'sample'.